src/matchers/to-have-prop.ts

```diff
@@ -0,0 +1,52 @@
+import type { ReactTestInstance } from 'react-test-renderer';
+import { matcherHint, stringify, printExpected } from 'jest-matcher-utils';
+import { checkReactElement, getMessage } from './utils';
+
+function printAttribute(name: string, value: unknown) {
+  return value === undefined ? name : `${name}=${stringify(value)}`;
+}
+
+function getPropComment(name: string, value: unknown) {
+  return value === undefined
+    ? `element.hasProp(${name})`
+    : `element.getAttribute(${name}) === ${stringify(value)}`;
+}
+
+export function toHaveProp(
+  this: jest.MatcherContext,
+  element: ReactTestInstance,
+  name: string,
+  expectedValue: unknown
+) {
+  checkReactElement(element, toHaveProp, this);
```

Comment:
You should use the already existing method `checkHostElement` and delete this one

Reply:
Yes, `checkHostElement` is the check to use